File: life-journey-backend/app/schemas/helpdesk.py
from pydantic import BaseModel, Field


class HelpdeskMessage(BaseModel):
    role: str  # "user" | "assistant"
    content: str


class HelpdeskChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=1000)
    conversation_history: list[HelpdeskMessage] = Field(default_factory=list)


class HelpdeskActionLink(BaseModel):
    label: str
    href: str


class HelpdeskChatResponse(BaseModel):
    message: str
    escalate: bool
    suggested_questions: list[str]
    action_links: list[HelpdeskActionLink]

File: life-journey-frontend/src/lib/api/helpdesk.ts
const API_BASE = process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:8001/api/v1";

function getToken(): string | null {
  if (typeof window === "undefined") return null;
  const raw = localStorage.getItem("life-journey.auth");
  if (!raw) return null;
  try {
    return JSON.parse(raw).token ?? null;
  } catch {
    return null;
  }
}

export interface HelpdeskMessage {
  role: "user" | "assistant";
  content: string;
}

export interface HelpdeskActionLink {
  label: string;
  href: string;
}

export interface HelpdeskResponse {
  message: string;
  escalate: boolean;
  suggested_questions: string[];
  action_links: HelpdeskActionLink[];
}

export async function sendHelpdeskMessage(
  message: string,
  conversationHistory: HelpdeskMessage[]
): Promise<HelpdeskResponse> {
  const token = getToken();
  const res = await fetch(`${API_BASE}/helpdesk/chat`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      ...(token ? { Authorization: `Bearer ${token}` } : {}),
    },
    body: JSON.stringify({
      message,
      conversation_history: conversationHistory,
    }),
  });

  if (!res.ok) {
    throw new Error("Kon je bericht niet versturen. Probeer het opnieuw.");
  }

  return res.json();
}
